src/components/Waypoints/WaypointMenu.tsx

```diff
@@ -0,0 +1,126 @@
+import React from "react"; //,  useState, useEffect, useCallback
+import { useSelector, useDispatch } from "react-redux";
+
+import {
+  XMarkIcon,
+  DocumentDuplicateIcon,
+  MapPinIcon,
+  PencilSquareIcon,
+  TrashIcon,
+} from "@heroicons/react/24/outline";
+
+import { useDeleteWaypoint, useToggleEditWaypoint } from "@app/utils/waypoint";
+import { writeValueToClipboard } from "@utils/clipboard";
+import { selectActiveWaypoint } from "@app/features/device/deviceSelectors";
+import { deviceSliceActions } from "@features/device/deviceSlice";
+
+// This file contains the WaypointMenu component when it is not being edited
+// It is called in MapView.tsx
+
+const WaypointMenu = () => {
+  const dispatch = useDispatch();
+  const activeWaypoint = useSelector(selectActiveWaypoint());
+
+  // Waypoint information
+  const waypointTitle = activeWaypoint?.name;
+  const waypointDescription = activeWaypoint?.description;
+  const waypointLong = activeWaypoint?.longitudeI
+    ? (activeWaypoint.longitudeI / 1e7).toString()
+    : null;
+  const waypointLat = activeWaypoint?.latitudeI
+    ? (activeWaypoint.latitudeI / 1e7).toString()
+    : null;
+
+  // Waypoint Utils, used when the buttons at bottom are clicked
+  const deleteWaypoint = useDeleteWaypoint();
+  const toggleEditWaypoint = useToggleEditWaypoint();
+
+  // Only show if there is an active waypoint
+  if (!activeWaypoint) {
+    return <></>;
+  } else {
```

Comment:
Nit: don't need an explicit `else` block since the `if` statement returns, meaning the only time we reach any code below the `if` statement is when `activeWaypoint` is not `null`